extensions/adobe/experience/customerJourneyManagement/message-experienceevent.schema.json

```diff
@@ -0,0 +1,114 @@
+{
+  "meta:license": [
+    "Copyright 2018 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/experience/customerJourneyManagement/message-experienceevent",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Adobe CJM ExperienceEvent Template Mixin",
+  "type": "object",
+  "description": "Adobe CJM ExperienceEvent mixin for ingesting the ExperienceEvent. Includes the core/standard ExperienceEvent as well as the required Adobe CJM ExperienceEvent mixins.",
+  "meta:extensible": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "meta:extends": ["https://ns.adobe.com/xdm/context/experienceevent"],
+  "definitions": {
+    "messageExperienceevent": {
+      "properties": {
+        "https://ns.adobe.com/experience/customerJourneyManagement/messageExecution": {
+          "title": "Message Execution",
+          "$ref": "https://ns.adobe.com/experience/customerJourneyManagement/messageexecution",
+          "description": "MessageExecution details for the Adobe CJM ExperienceEvent."
+        },
+        "https://ns.adobe.com/experience/customerJourneyManagement/messageProfile": {
+          "title": "Message Profile",
+          "$ref": "https://ns.adobe.com/experience/customerJourneyManagement/messageprofile",
+          "description": "MessageProfile details for the Adobe CJM ExperienceEvent."
+        },
+
+        "https://ns.adobe.com/experience/customerJourneyManagement/channelContext": {
+          "title": "Message Channel context",
+          "description": "Message Channel fields for the Messageprofle associated with the Adobe CJM Experience Event.",
+          "properties": {
+            "https://ns.adobe.com/experience/customerJourneyManagement/emailChannelContext": {
+              "title": "Email Channel Context",
+              "$ref": "https://ns.adobe.com/experience/customerJourneyManagement/email-channel-context",
+              "description": "Email channel fields for the Messageprofle associated with the Adobe CJM Experience Event."
+            },
+            "https://ns.adobe.com/experience/customerJourneyManagement/pushChannelContext": {
+              "title": "Push Channel Context",
+              "$ref": "https://ns.adobe.com/experience/customerJourneyManagement/push-channel-context",
+              "description": "Push channel fields for the MessageProfile associated with the Adobe CJM Experience Event."
+            }
+          },
+           "oneOf": [
```

Comment:
I'm checking, but pretty sure the AEP ingestion process will not support this. "oneOf" is a json-schema concept not really supported in AEP at this time.

Reply:
I checked and have confirmed that this oneOf around the "required" attribute is not supported by the ingestion process. There is no way today to say you are required to have field X OR Y. If this is a problem, then please open a ticket with the data ingestion team.

Reply:
Thanks @cmathis for update, I have removed the use of "oneOf" construct for defining the required properties.

Reply:
@cmathis : You can resume the review. Also please provide your inputs regarding comment https://github.com/adobe/xdm/pull/915#discussion_r469872849.

We wanted to commit "message-experienceevent.schema" as a schema(not mixin which needs to be added over EE class) so that it can be available as OOTB schema in AEP.

If we have to make any changes for that, we will incorporate that.